File: ganilib-core/src/main/java/com/gani/lib/select/SelectableItem.java
package com.gani.lib.select;

import android.view.View;
import android.widget.TextView;

import com.gani.lib.utils.string.StringUtils;

import java.io.Serializable;
import java.util.LinkedList;
import java.util.List;

public interface SelectableItem extends Serializable {
  String getName();

  

  public class Utils {
    private static List<String> collectNames(List<? extends SelectableItem> items) {
      List<String> itemNames = new LinkedList<String>();
      for (SelectableItem item : items) {
        itemNames.add(item.getName());
      }
      return itemNames;
    }

    public static String toLineSeparatedNames(List<? extends SelectableItem> items) {
      return StringUtils.join(collectNames(items), "\n");
    }

    public static String toSpaceSeparatedNames(List<? extends SelectableItem> items) {
      return StringUtils.join(collectNames(items), " ");
    }

    // Set to empty when there is no error. This is important for determining form validity (e.g. when enabling submit button).
    public static void initLayout(TextView selected, TextView error, List<? extends SelectableItem> items) {
      String names = toLineSeparatedNames(items);
      selected.setText(names);
      selected.setVisibility((names.isEmpty())? View.GONE : View.VISIBLE);
      
      if (names.isEmpty()) {
//        error.setText(R.string.error_empty);
        error.setText("Cannot be empty");
      }
      else {
        error.setText("");
      }
    }
  }
}

File: glib-core/src/main/java/com/glib/core/ui/select/SelectableItem.kt
package com.glib.core.ui.select

import android.view.View
import android.widget.TextView
import com.glib.core.utils.string.StringUtils.join
import java.io.Serializable
import java.util.*

interface SelectableItem : Serializable {
    val name: String

    object Utils {
        private fun collectNames(items: List<SelectableItem>): List<String> {
            val itemNames: MutableList<String> =
                LinkedList()
            for (item in items) {
                itemNames.add(item.name)
            }
            return itemNames
        }

        fun toLineSeparatedNames(items: List<SelectableItem>): String {
            return join(
                collectNames(
                    items
                ), "\n"
            )!!
        }

        fun toSpaceSeparatedNames(items: List<SelectableItem>): String {
            return join(
                collectNames(
                    items
                ), " "
            )!!
        }

        // Set to empty when there is no error. This is important for determining form validity (e.g. when enabling submit button).
        fun initLayout(
            selected: TextView,
            error: TextView,
            items: List<SelectableItem>
        ) {
            val names =
                toLineSeparatedNames(items)
            selected.text = names
            selected.visibility = if (names.isEmpty()) View.GONE else View.VISIBLE
            if (names.isEmpty()) {
//        error.setText(R.string.error_empty);
                error.text = "Cannot be empty"
            } else {
                error.text = ""
            }
        }
    }
}